### ISSUE TITLE
Inheriting enum from byte causes exeption.

### ISSUE THREAD

**Issue by MaartenThomassen** [NSwag/NSwag#30]
When setting enum handling to "Integer" and parsing an enum which inherits from byte, the code throws and exeption.

Example: 

```
public enum PermissionFlags : byte
    {
        Create  = 0x01,
        Read    = 0x02,
        Update  = 0x04,
        Delete  = 0x08,
    }
```

The exception occurs in NJsonSchema (line 237 of JsonScheamGenerator of NJsonSchema version NJsonSchema.1.18.5821.36806).  

They changed NJsonSchema code since that verion (version 1.22 is the most recent as of this post) and swapped around a bit of code, it is now on line 248 of that same file. I will post an issue with NJsonSchema and post a link to that later.

**Comment by MaartenThomassen** [comment]
See https://github.com/NJsonSchema/NJsonSchema/issues/19

**Comment by rsuter** [comment]
Fixed in v1.25
